src/main/java/com/synopsys/integration/blackduck/nexus3/task/metadata/MetaDataTask.java

```diff
@@ -196,71 +198,72 @@ protected void execute(final Repository repository) {
         }
     }
 
-    private void updatePendingScanAssets(final Map<String, AssetWrapper> assetWrapperToWaitFor, final CodeLocationCreationService codeLocationCreationService, final ProjectService projectService, final BlackDuckService blackDuckService) {
+    private void updatePendingScanAssets(Map<String, AssetWrapper> assetWrapperToWaitFor, CodeLocationCreationService codeLocationCreationService, ProjectService projectService, BlackDuckService blackDuckService) {
         NotificationTaskRange notificationTaskRange = null;
         CodeLocationWaitResult codeLocationWaitResult = null;
         String errorMessage = null;
         int timeout = -1;
         try {
             timeout = commonRepositoryTaskHelper.getBlackDuckServerConfig().getTimeout() * 5;
             notificationTaskRange = codeLocationCreationService.calculateCodeLocationRange();
-            final Set<String> codeLocationNames = assetWrapperToWaitFor.keySet();
+            Set<String> codeLocationNames = assetWrapperToWaitFor.keySet();
             codeLocationWaitResult = codeLocationCreationService.waitForCodeLocations(notificationTaskRange, codeLocationNames, codeLocationNames.size(), timeout);
-        } catch (final InterruptedException e) {
+        } catch (InterruptedException e) {
             errorMessage = "Waiting for the scan to complete was interrupted: " + e.getMessage();
-            logger.error("Problem communicating with Black Duck: {}", errorMessage);
-        } catch (final BlackDuckApiException e) {
+            logger.error(errorMessage);
+            Thread.currentThread().interrupt();
```

Comment:
Do we need to interrupt the thread if the Exception is an InterruptedException?

Reply:
Since we are catching the InterruptedException we have to do the thread interrupt since we are not re-throwing the exception.